### PR TITLE
Moved Community section of the index to its own topic

### PR DESCRIPTION
**Pull request by barbaricyawps**
### What does this PR do?
This is a minor documentation change for organization purposes. The "Introduction to Slack" topic included a discussion of Slack's features and then ended with a section discussing ways to join the Salt community. I felt it would make more sense to break out the Salt Community section into its own separate topic . I also added the new topic to the Table of Contents.

### What issues does this PR fix or reference?
This will help make your open source documentation more clear and well-organized.

### Tests written?
No

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by garethgreenaway — APPROVED** [review]

**Review by rallytime — APPROVED** [review]
Thank you @barbaricyawps  and welcome!

**Comment by rallytime** [comment]
Hi @barbaricyawps -  I think because I merged in your other PR, that caused conflicts with this change. Can you rebase this with the current HEAD of `develop` and fix the conflicts?

**Comment by cachedout** [comment]
@rallytime and @barbaricyawps I went ahead and resolved the merge conflict so we should be all set to go here.

File: doc/contents.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Salt Table of Contents
     :maxdepth: 2
 
     topics/index
+	topic/community
```
Comment (rallytime) [inline doc/contents.rst]: I don't think this is _quite_ right. I just built the docs with the changes in this PR. While I can access the new community page via the direct link, I don't see it on the table of contents.

There is also a related warning generated when building the docs:
```
/root/SaltStack/salt/doc/contents.rst:7: WARNING: toctree contains reference to nonexisting document u'    topic/community'
```

**Review by rallytime — CHANGES_REQUESTED** [review]
HI @barbaricyawps - Can you take another look here? There is an issue with building the docs that needs to be resolved before we can merge this.

**Comment by barbaricyawps** [comment]
So, to clarify, you want me to pull the latest version of the repository, resolve the conflicts, then resubmit the changes? (I'm sorry, I'm a Mercurial person, so I get the concepts, but I don't always know how to do it in Github.)

File: doc/contents.rst (hunk 1 above)
Comment (terminalmage) [reply doc/contents.rst]: @barbaricyawps You just need to align this with `topics/index`. Also, this should be `topics/community/index` and not `topic/community`.

**Review by terminalmage — CHANGES_REQUESTED** [review]

**Comment by rallytime** [comment]
Hi @barbaricyawps - did you have a chance to follow up on the feedback here? We'd love to get this merged in.

**Comment by cachedout** [comment]
Fixed the indentation issue and re-running tests.

**Comment by rallytime** [comment]
This still has some trouble with the doc build warnings, but i will fix them up in a subsequent PR today.

**Pull request merged**